plugin/trino-bigquery/src/test/java/io/trino/plugin/bigquery/TestBigQueryIntegrationSmokeTest.java

```diff
@@ -244,6 +244,21 @@ public void testSelectFromYearlyPartitionedTable()
         assertEquals((long) actualValues.getOnlyValue(), 1L);
     }
 
+    @Test(description = "regression test for https://github.com/trinodb/trino/issues/7784")
+    public void testSelectWithSingleQuoteInWhereClause()
+    {
+        String tableName = "test.select_with_single_quote";
+
+        onBigQuery("DROP TABLE IF EXISTS " + tableName);
+        onBigQuery("CREATE TABLE " + tableName + "(col INT64, val STRING)");
+        onBigQuery("INSERT INTO " + tableName + " VALUES (1,'Looney\\'s Lane')");
+
+        MaterializedResult actualValues = computeActual("SELECT val FROM " + tableName + " where val='Looney''s Lane'");
```

Comment:
Uppercase `WHERE` and add space `value = `.

Reply:
Done